### DISCUSSION TITLE
CSLA.Net & Syncfusion Blazor Grid

### DISCUSSION THREAD

**Discussion by ashishsinha24** [MarimerLLC/csla#2088] [Questions]
What is the proper way to attach to a blazor datagrid?

I am using a stripped down version of the ProjectTracker Sample and I am attaching the sample project where I have attached the grid to the project menu. I found out that if there are more than one record in the table, it doesnt show anything. But if I set it to a single record, the syncfusion grid starts displayin data.

What am I missing?
[ProjectTracker.zip](https://github.com/MarimerLLC/csla/files/5926512/ProjectTracker.zip)

**Comment by rockfordlhotka** [comment]
I don't have the bandwidth to troubleshoot every third party control suite out there. Just like with Windows Forms and WPF, if there's a question with a built-in Microsoft control I'll do my best, but third party controls have to be community supported.

**Comment by uwefms** [comment]
@ashishsinha24,

  if you still having your problem try this ...

[Syncf_ProjectTracker.zip](https://github.com/MarimerLLC/csla/files/6177264/Syncf_ProjectTracker.zip)

Sorry for the delay - just read about your problem today.

    regards

         Uwe

**Comment by kcabral817** [comment]
I think there is a problem with the sycfusion binding engine that won't bind correctly to the properties of csla object.  Syncfusion works when using a simple poco, but from what I can see including the sync_projectTracker is it becomes problematic.  I would call attention to the ".toList()" when binding a CSLA listobject to the grid.  I'm still working through this but syncfusion tech support is of no use and keep referring back to simple design.  It reminds me of the pharmaceutical industry where the drug works in the lab and fails miserably in human trials.  It don't matter how well it works in the lab, in the real world the control has to work in all situations when binding to a valid object.

**Reply by rockfordlhotka** [reply]
Who is calling `ToList`? The datagrid control?

**Comment by kcabral817** [comment]
The .toList() is in an example from a person who put CSLA and SFGrid together in a demo application.  I am finding that the problem is with the invocation of the class's parameterless constructor.  I am having to intercept the call so that I can create the csla object properly.  I tried both syncfusion and telerik's blazor data grids and they both behave the same.
